corehq/apps/fixtures/upload/run_upload.py

```diff
@@ -45,10 +45,22 @@ def _run_upload(domain, workbook, replace=False, task=None):
     records to optimize database interactions.
     """
     def process_table(table):
-        new_rows = workbook.iter_rows(table)
+        def process_row(row):
+            if not replace:
+                uid = workbook.get_uid(row)
+                if uid in rows_by_id:
+                    rows.to_delete.append(rows_by_id[uid])
+
         old_rows = FixtureDataItem.get_item_list(domain, table.tag)
-        mutation = get_mutation(old_rows, new_rows, row_key)
-        rows.update(mutation)
+        rows_by_id = {r._id: r for r in old_rows}
+        assert None not in rows_by_id, rows_by_id[None]
```

Comment:
Yeah, sorry I understand how assertions are used. I meant that we often use formatted strings for the assertion message (you can see examples of that in sentry now), which makes it easier to understand what the assertion was supposed to mean in the unlikely event there is a bug and the assertion is triggered.

Reply:
Sure, I think messages in an assertion like https://github.com/dimagi/commcare-hq/blob/master/corehq/util/doc_processor/couch.py#L167 make it a bit easier to see exactly what is wrong, especially for someone other than the author of the assertion, since they describe the issue rather than just showing the `repr` of a `FixtureDataItem` class, which may not be immediately obvious to someone else who finds that error in sentry.

This is definitely not a blocker, but I do think it's a nice convention that shows up frequently in our code and makes it a lot easier if someone other than the original author stumbles upon an `AssertionError`